### PR TITLE
cmd/sniproxy: implement support for control configuration, multiple addresses

### PR DESCRIPTION
**Pull request by twitchyliquid64**
* Implement missing tests for sniproxy
 * Wire sniproxy to new appc package
 * Add support to tsnet for routing subnet router traffic into netstack, so it can be handled

Updates: https://github.com/tailscale/corp/issues/15038

### CODE REVIEW

**Comment by twitchyliquid64** [comment]
Nb: Ive heard chatter around `tsnet.HandleOtherAddresses` being the wrong knob, so I'll revisit that. Should be okay for the rest of the PR to be reviewed tho.

File: cmd/sniproxy/sniproxy_test.go [hunk 1]
```diff
@@ -35,3 +55,166 @@ func TestPortForwardingArguments(t *testing.T) {
 		}
 	}
 }
+
+var verboseDERP = flag.Bool("verbose-derp", false, "if set, print DERP and STUN logs")
+var verboseNodes = flag.Bool("verbose-nodes", false, "if set, print tsnet.Server logs")
```
Comment (catzkorn) [inline cmd/sniproxy/sniproxy_test.go:60]: 👁️ 👍🏻

File: cmd/sniproxy/sniproxy_test.go [hunk 2]
```diff
@@ -35,3 +55,166 @@ func TestPortForwardingArguments(t *testing.T) {
...
+				IP: []tailcfg.ProtoPortRange{
+					{
+						Proto: int(ipproto.TCP),
+						Ports: tailcfg.PortRange{First: uint16(ln.Addr().(*net.TCPAddr).Port), Last: uint16(ln.Addr().(*net.TCPAddr).Port)},
+					},
+				},
+			},
+		},
+	}
+	b, _ := json.Marshal(config)
```
Comment (catzkorn) [inline cmd/sniproxy/sniproxy_test.go:144]: I think we should check the error here - just to be safe.

**Comment by twitchyliquid64** [comment]
As resolved in chat: we can just set `netstack.ProcessSubnets` to true for tsnet unconditionally, because there isnt a valid usecase for tsnet binaries routing packets as a subnet router.

I've pulled out the `tsnet.Server.HandleOtherAddresses` knob.

**Comment by twitchyliquid64** [comment]
tsnet change has been moved to own PR: https://github.com/tailscale/tailscale/pull/9958

File: cmd/sniproxy/sniproxy.go [hunk 3]
```diff
@@ -10,30 +10,34 @@ package main
...
 	"tailscale.com/tsnet"
 	"tailscale.com/tsweb"
+	"tailscale.com/types/appctype"
+	"tailscale.com/types/ipproto"
 	"tailscale.com/types/nettype"
-	"tailscale.com/util/clientmetric"
+	"tailscale.com/util/mak"
 )
 
+const configCapKey = "tailscale.com/appconnector"
```
Comment (raggi) [inline cmd/sniproxy/sniproxy.go:39]: did we wanna update this to /sniproxy?

**Review by raggi — APPROVED** [review]

File: cmd/sniproxy/sniproxy.go (hunk 3 above)
Comment (twitchyliquid64) [reply cmd/sniproxy/sniproxy.go]: Done!

**Pull request merged**